ThisAssemblyNamespace is ignored when updating from 2.3.0 to 3.0.4

## Describe the Bug

error CS0436: The type 'ThisAssembly' in 'GitInfo.CodeAnalysis\GitInfoGenerator\ThisAssembly.Git.IsDirty.g.cs' conflicts with the imported type 'ThisAssembly' in 'myassembly, Culture=neutral, PublicKeyToken=...'. Using the type defined in 'GitInfo.CodeAnalysis\GitInfoGenerator\ThisAssembly.Git.IsDirty.g.cs'.

## Steps to Reproduce

I updated my project from 2.3.0 to 3.0.4 and run into this issue. Using 3.0.0. results in 
    
    C:\Program Files\dotnet\sdk\7.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.GenerateAssemblyInfo.targets(209,5): error NETSDK1018: Invalid NuGet version string: '123.45.678+<$branchname>.<$commitid>'. 

Note: Branchname contains ´/´, and gitversion shows `v123.45-678-g<$commitid>` instead of `123.45.678`.

## Expected Behavior

Updating to 3.0.4 should not break a project that was working with 2.3.0.

## Version Info

working version: 2.3.0
breaking versions: 3.0.4 (and 3.0.0)

## Additional Info

I tried to create a [minimal sample](https://github.com/codingdave/GitInfoTest) but wasnt able to reproduce the issue over there, though I tried to create the same setup is to my failing project. 

So I hope you can give me some hints how to continue.

--- a/src/GitInfo/build/GitInfo.AssemblyMetadata.targets
+++ b/src/GitInfo/build/GitInfo.AssemblyMetadata.targets
@@ -10,12 +10,7 @@
 	==============================================================
 	-->
 
-  <PropertyGroup>
-    <GitThisAssemblyMetadata Condition="'$(GitThisAssemblyMetadata)' == ''">false</GitThisAssemblyMetadata>
-  </PropertyGroup>
-
-  <Target Name="GitAssemblyMetadata" DependsOnTargets="GitInfo;GitVersion"
-          BeforeTargets="BuildOnlySettings" Condition="'$(GitThisAssemblyMetadata)' == 'true'">
+  <Target Name="GitAssemblyMetadata" DependsOnTargets="GitInfo;GitVersion" BeforeTargets="BuildOnlySettings">
 
     <ItemGroup Condition="'$(Language)' != 'VB'">
       <AssemblyMetadata Include="GitInfo.IsDirty"

--- a/src/GitInfo/build/GitInfo.targets
+++ b/src/GitInfo/build/GitInfo.targets
@@ -1074,9 +1074,12 @@
     </PropertyGroup>
   </Target>
 
-  <Import Project="GitInfo.AssemblyInfo.targets"/>
-  <Import Project="GitInfo.AssemblyMetadata.targets" Condition="'$(Language)' != 'C#' or '$(ThisAssemblyNamespace)' != ''"/>
-  <Import Project="GitInfo.ThisAssembly.targets" Condition="'$(Language)' == 'C#'"/>
+  <Import Project="GitInfo.AssemblyMetadata.targets" Condition="'$(GitThisAssemblyMetadata)' == 'true'"/>
+
+  <!-- Legacy generation in place for non-C# or when the unsupported ThisAssemblyNamespace is in use -->
+  <Import Project="GitInfo.AssemblyInfo.targets" Condition="'$(Language)' != 'C#' or '$(ThisAssemblyNamespace)' != ''"/>
+  <!-- Otherwise, for C# we always use ThisAssembly instead. -->
+  <Import Project="GitInfo.ThisAssembly.targets" Condition="'$(Language)' == 'C#' and '$(ThisAssemblyNamespace)' == ''"/>
 
   <PropertyGroup>
     <GitInfoImported>true</GitInfoImported>

